docs/DevGuide/Parallelization.md

```diff
@@ -0,0 +1,539 @@
+
+Each %Parallel Component struct must have the following type aliases:
+1. `using chare_type` is set to one of:
+   1. `Parallel::Algorithms::Singleton`s have one object in the entire execution
+      of the program.
+   2. `Parallel::Algorithms::Array`s hold zero or more elements, each of which
+      is an object distributed to some core. An array can grow and shrink in
+      size dynamically if need be and can also be bound to another array. A
+      bound array has the same number of elements as the array it is bound to,
+      and elements with the same ID are on the same core. See Charm++'s chare
+      arrays for details.
+   3. `Parallel::Algorithms::Group`s are arrays with
+      one element per core which are not able to be moved around between
+      cores. These are typically useful for gathering data from array elements
+      on their core, and then processing or reducing the data further. See
+      [Charm++'s](http://charm.cs.illinois.edu/help) group chares for details.
+   4. `Parallel::Algorithms::Nodegroup`s are similar to
+      groups except that there is one element per node. For Charm++ SMP (shared
+      memory parallelism) builds, a node corresponds to the usual definition of
+      a node on a supercomputer. However, for non-SMP builds nodes and cores are
+      equivalent. We ensure that all entry method calls done through the
+      Algorithm's `simple_action` and `receive_data` functions are
+      threadsafe. User controlled threading is possible by calling the non-entry
+      method member function `threaded_action`.
+2. `using metavariables` is set to the Metavariables struct that stores the
+   global metavariables. It is often easiest to have the %Parallel
+   Component struct have a template parameter `Metavariables` that is the
+   global metavariables struct. Examples of this technique are given below.
+3. `using phase_dependent_action_list` is set to a `tmpl::list` of
+   `Parallel::PhaseActions<PhaseType, Phase, tmpl::list<Actions...>>` where each
+   `PhaseAction` represents a PDAL that will be executed on the parallel
+   component during the specified phase. The  %Actions are executed in the order
+   that they are given in the `tmpl::list`s of the PDALs, but the phases need
+   not be run in linear order. However, `DataBox` types are constructed assuming
+   the phases are performed from first in the `phase_dependent_action_list` to
+   the last. Simple actions (described below) can be executed in any phase. If
+   there are no iterable actions in a phase then a `PhaseAction` need not be
+   specified for that phase. However, at least one `PhaseAction`, even it if is
+   empty, must be specified.
+4. `using options` is set to a (possibly empty) `tmpl::list` of the option
+   structs. The options are read in from the input file specified in the main
+   `Metavariables` struct. After being read in they are passed to the
+   `initialize` function of the parallel component, which is described below.
```

Comment:
suggestion (unless the statement is false):  These options are only used during initialization.

Reply:
Not strictly required, but I think it will be generally true. I'll add that :)